app/src/main/java/com/mobileproto/hireddit/hireddit/GetWordsAsync.java

```diff
@@ -0,0 +1,120 @@
+package com.mobileproto.hireddit.hireddit;
+
+import android.content.Context;
+import android.content.res.AssetManager;
+import android.os.AsyncTask;
+import android.util.Log;
+import android.widget.TextView;
+
+import com.google.common.io.CharStreams;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
+import io.indico.Indico;
+import io.indico.network.IndicoCallback;
+import io.indico.results.IndicoResult;
+import io.indico.utils.IndicoException;
+
+;
+
+/**
+ * Created by lwilcox on 11/5/2015.
+ */
+public class GetWordsAsync extends AsyncTask<Void, Void, ArrayList<String>>
+{
+    private String spokenString;
+    private String importantWords;
+    private Context context;
+    public ArrayList<String> wordList = new ArrayList<>();
+    public ArrayList<String> allComments;
+    public String postComment;
+    public TextView commentText;
+    public Indico indico;
+
+
+    public GetWordsAsync(String spokenString, String importantWords, Context context, TextView commentText)
+    {
+
+        this.spokenString = spokenString;
+        this.importantWords = importantWords;
+        this.context = context;
+        this.commentText = commentText;
+        String indicoApiKey = getApi(context);
+        this.indico = Indico.init(context, indicoApiKey, null);
+
+    }
+
+    public static String getApi(Context context)
+    {
+        try {
+            AssetManager assetManager = context.getAssets();
+            InputStream apikey = assetManager.open("indicoapitxt.txt");
+            String apiKeyString = CharStreams.toString(new InputStreamReader(apikey, "UTF-8"));
+            Log.d("apikey", apiKeyString);
+            return apiKeyString;
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    @Override
+    protected ArrayList<String> doInBackground(Void... params)
+    {
+        try {
+            //final ArrayList<String> wordList = new ArrayList<String>();
+            indico.keywords.predict(spokenString, new IndicoCallback<IndicoResult>()
+            {
+                @Override
+                public void handle(IndicoResult result) throws IndicoException
+                {
+                    Log.i("Indico Keywords", "keywords: " + result.getKeywords());
+                    if (result.getKeywords() != null) {
+                        wordList.add(result.getKeywords().keySet().toString());
+                    }
+                }
+            });
+            Log.d("wordlist", wordList.toString());
+            while (wordList.isEmpty()) {
+                Log.d("wating", "true");
+                try {
+                    Thread.sleep(15);
```

Comment:
I'd like to try fixing this issue before merging it into master. Could we
address all the other comments in this PR first and I'll try to find time
to look at this sleeping thingy

On Mon, Nov 16, 2015 at 2:42 PM Lucy Wilcox <notifications@github.com>
wrote:

> In app/src/main/java/com/mobileproto/hireddit/hireddit/GetWordsAsync.java
> <https://github.com/greneroom/Hi-Reddit/pull/20#discussion_r44971855>:
>
> > +            indico.keywords.predict(spokenString, new IndicoCallback<IndicoResult>()
> > +            {
> > +                @Override
> > +                public void handle(IndicoResult result) throws IndicoException
> > +                {
> > +                    Log.i("Indico Keywords", "keywords: " + result.getKeywords());
> > +                    if (result.getKeywords() != null) {
> > +                        wordList.add(result.getKeywords().keySet().toString());
> > +                    }
> > +                }
> > +            });
> > +            Log.d("wordlist", wordList.toString());
> > +            while (wordList.isEmpty()) {
> > +                Log.d("wating", "true");
> > +                try {
> > +                    Thread.sleep(15);
>
> It's not that indico is failing, it's just something with the time it
> takes to actually make the call. I think we should merge this and then
> create an issue and fix it separately.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/greneroom/Hi-Reddit/pull/20/files#r44971855>.
>